lib/collection/src/operations/verification/mod.rs

```diff
@@ -0,0 +1,160 @@
+mod search;
+
+use std::fmt::Display;
+
+use segment::types::Filter;
+
+use super::config_diff::StrictModeConfig;
+use crate::collection::Collection;
+
+/// Trait to verify strict mode for requests. All functions in this trait will default to an 'empty' implementation,
+/// which means every value that needs to be checked must be implemented.
+/// This trait ignores the `enabled` parameter in `StrictModeConfig`.
+pub trait StrictModeVerification {
+    /// Implementing this method allows adding a custom check for request specific values.
+    fn check_custom(
+        &self,
+        _collection: &Collection,
+        _strict_mode_config: &StrictModeConfig,
+    ) -> Result<(), String> {
```

Comment:
I think we could either add variants in `CollectionError` or create a new kind of error implement conversion into `CollectionError`